homeassistant/components/stream/recorder.py

```diff
@@ -25,12 +25,23 @@ def recorder_save_worker(file_out: str, segments: List[Segment], container_forma
     output_v = None
     output_a = None
 
-    for segment in segments:
-        # Seek to beginning and open segment
-        segment.segment.seek(0)
+    # Get first_pts values from first segment
+    if len(segments) > 0:
+        segment = segments[0]
         source = av.open(segment.segment, "r", format=container_format)
         source_v = source.streams.video[0]
+        first_pts["video"] = source_v.start_time
+        if len(source.streams.audio) > 0:
+            source_a = source.streams.audio[0]
+            first_pts["audio"] = int(
+                source_v.start_time * source_v.time_base / source_a.time_base
+            )
+        source.close()
 
+    for segment in segments:
+        # Open segment
+        source = av.open(segment.segment, "r", format=container_format)
```

Comment:
Does this work? I believe the reason I had to add the seek in the first place is because after write/close the pointer in the `BytesIO` was at the end, so when calling `av.open` it didn't read the stream properly.

Reply:
Hmm, this seems to work on my end. Maybe you can test it out when you get a chance? That said, there does seem to be a fair amount of seemingly random stuff we have in the component that's actually there for good reason :-)

Reply:
It's possible it was added back when I wasn't closing the output container (which PyAV didn't allow you to do until a recent version bump). Now that we are, it's entirely possible that the act of closing the stream resets it automatically.